### PR TITLE
Remove more explicit references to SearchResponse in tests (#101092)

### PR DESCRIPTION
**Pull request by original-brownbear**
Follow-up to #100966.

Add more assertion overloads that consume a requestBuilder as in the other PRs and start using `assertHitCount` in more places that were duplicating what it does.
Also add a shortcut for `client().prepareSearch()` to integ tests and bulk-replace some usages of this pattern to avoid these changes from blowing up test code line count further.

### CODE REVIEW

**Review by DaveCTurner — COMMENTED** [review]
> bulk-replace some usages of this pattern

TBH bulk-replacing them all would make this easier to review, by just repeating the exercise locally :) See https://github.com/elastic/elasticsearch/pull/101175.

**Review by DaveCTurner — COMMENTED** [review]
Didn't spot any obvious problems but let's see how it looks when combined with #101175. One comment about checked exceptions anyway.

File: test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java [hunk 1]
```diff
@@ -305,6 +306,16 @@ public static void assertHitCount(SearchRequestBuilder searchRequestBuilder, lon
         }
     }
 
+    public static void assertHitCount(ActionFuture<SearchResponse> responseFuture, long expectedHitCount) throws ExecutionException,
+        InterruptedException {
+        var res = responseFuture.get();
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java]: Eh that's a good point. Maybe an unchecked exception is better. So trappy to have `AssertionError` mean "try again" sometimes. I'll defer to your judgement here.

[the remark this answers is not in the document]
Comment (original-brownbear) [reply test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java]: Hmm thought this through some more, it's really unlikely that being tricky here with re throwing and unwrapping in some test actually adds much value here. I went with keeping it simple and throwing assertion error now :)

**Review by DaveCTurner — COMMENTED** [review]
LGTM

(darnit my search-and-replace missed all the zero-arg usages)

**Review by DaveCTurner — APPROVED** [review]

**Pull request merged**